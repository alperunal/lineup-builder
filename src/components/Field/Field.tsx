import React from 'react';
import Player from '../Player/Player';
import { IPlayer } from '../../constants/model';
import './Field.scss';

interface IProps {
    width: number;
    height: number;
    players: IPlayer[];
    mainColor: string;
    secondaryColor: string;
    numberColor: string;
    move: (event: any) => void;
    drop: (event: React.DragEvent) => void;
    handleDragStart: (event: React.DragEvent) => void;
}

const Field: React.FC<IProps> = ({ width, height, players, mainColor, secondaryColor, numberColor, move, drop, handleDragStart }: IProps) => {
    return (
        <div
            className="field"
            style={{
                maxHeight: `${height}px`,
                maxWidth: `${width}px`,
            }}
            onDrop={drop}
            onDragOver={(event: React.DragEvent) => event.preventDefault()}
            onMouseMove={move}
        >
            <div className="field__field-lines">
                <div className="field__grass-texture" />
                <div
                    className="field__goal-line"
                    style={{
                        top: '-2px',
                    }}
                />
                <div
                    className="field__goal-line"
                    style={{
                        bottom: '-2px'
                    }}
                />
                <div
                    className="field__penalty-line"
                    style={{
                        borderTop: 'none',
                        top: 0,
                    }}
                />
                <div
                    className="field__penalty-line"
                    style={{
                        borderBottom: 'none',
                        bottom: 0,
                    }}
                />
                <div className="field__middle-line" />
                <div className="field__middle-circle" />
            </div>
            <div className="field__drag-layer" />
            {players.map(player => (
                <Player
                    key={player.id}
                    id={player.id}
                    name={player.name}
                    num={player.num}
                    x={player.position.x}
                    y={player.position.y}
                    mainColor={mainColor}
                    secondaryColor={secondaryColor}
                    numberColor={numberColor}
                    handleDragStart={handleDragStart}
                />
            ))}
        </div>
    );
};

export default Field;
