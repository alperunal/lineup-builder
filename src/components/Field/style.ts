import styled from 'styled-components';

interface IWrapperProps {
    w: number;
    h: number;
}
  
export const Wrapper = styled.div`
  position: relative;
  height: 500px;
  max-height: ${(props: IWrapperProps) => `${props.h}px`};
  width: 100%;
  max-width: ${(props: IWrapperProps) => `${props.w}px`};
  background-color: #1b7e45;

  @media (min-width: 720px) {
    height: ${(props: IWrapperProps) => `${props.h}px`};
  }
`;

export const DragLayer = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
`;

export const GrassTexture = styled.div`
  position: absolute;
  background-repeat: repeat;
  background-size: 75px 75px;
  background-position: -20px -20px;
`;

export const FieldLines = styled.div`
  width: 99%;
  height: 99%;
  border: solid 3px rgba(255, 255, 255, 0.5);
  position: relative;
`;

export const GoalLine = styled.div`
  bottom: ${({ bottom }: { bottom?: boolean }) => (bottom ? "-2px" : "auto")};
  top: ${({ bottom }: { bottom?: boolean }) => (bottom ? "auto" : "-2px")};
  width: 16%;
  height: 6%;
  border: solid 3px rgba(255, 255, 255, 0.5);
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
`;

export const PenaltyLine = styled.div`
  position: absolute;
  width: 44%;
  height: 14%;
  border: solid 3px rgba(255, 255, 255, 0.5);
  border-bottom: ${({ bottom }: { bottom?: boolean }) =>
    bottom ? "none" : ""};
  border-top: ${({ bottom }: { bottom?: boolean }) => (bottom ? "" : "none")};
  left: 0;
  right: 0;
  margin: auto;
  bottom: ${({ bottom }: { bottom?: boolean }) => (bottom ? "0" : "auto")};
  top: ${({ bottom }: { bottom?: boolean }) => (bottom ? "auto" : "0")};
`;

export const MiddleLine = styled.div`
  position: absolute;
  box-sizing: border-box;
  width: 100%;
  height: 0;
  border: solid 2px rgba(255, 255, 255, 0.5);
  top: 50%;
  transform: translateY(-50%);
  left: 0;
  right: 0;
`;

export const MiddleCircle = styled.div`
  position: absolute;
  box-sizing: border-box;
  width: 20%;
  height: 16%;
  border: solid 3px rgba(255, 255, 255, 0.5);
  top: 50%;
  transform: translateY(-50%);
  left: 0;
  right: 0;
  margin: auto;
  border-radius: 50%;
`;