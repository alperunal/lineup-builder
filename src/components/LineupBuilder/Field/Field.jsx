import React from 'react';
import PropTypes from 'prop-types';

import Player from 'components/LineupBuilder/Player/Player';
import { PositionArea } from 'constants/position';

import grassTile from 'assets/tiles/grass-tile.png';
import wetGrassTile from 'assets/tiles/wet-grass-tile.png';
import dryGrassTile from 'assets/tiles/dry-grass-tile.png';

import './Field.module.scss';

function getFieldStyle(fieldType) {
    switch (fieldType) {
    case 'hard':
        return 'none';
    case 'grass':
        return `url('${grassTile}')`;
    case 'dry-grass':
        return `url('${dryGrassTile}')`;
    case 'wet-grass':
        return `url('${wetGrassTile}')`;
    default:
        return `url('${grassTile}')`;
    }
}

const Field = ({
    width, height, players, mainColor, secondaryColor, numberColor, id, fieldType, setPlayerPosition, showPositionAreas,
}) => {
    const positionList = [
        new PositionArea('ST', '#4d4dff'),
        new PositionArea('WL', '#a862ea'),
        new PositionArea('WR', '#ff4dff'),
        new PositionArea('AMC', '#ff6680'),
        new PositionArea('ML', '#ff4d4d'),
        new PositionArea('MC', '#ea9a62'),
        new PositionArea('MR', '#ffff4d'),
        new PositionArea('DMC', '#c2c2ff'),
        new PositionArea('WBL', '#e1c9f8'),
        new PositionArea('FBL', '#ffc2ff'),
        new PositionArea('DC', '#ffb3bf'),
        new PositionArea('WBR', '#ffbebe'),
        new PositionArea('FBR', '#f6dbc6'),
        new PositionArea('GK', '#ffe7be'),
    ];

    return (
        <div
            className="field"
            style={{
                maxHeight: `${height}px`,
                maxWidth: `${width}px`,
                backgroundImage: getFieldStyle(fieldType),
            }}
            id={id}
        >
            <div className="field__field-lines">
                <div className="field__grass-texture" />
                <div
                    className="field__goal-line"
                    style={{
                        top: '-2px',
                    }}
                />
                <div
                    className="field__goal-line"
                    style={{
                        bottom: '-2px',
                    }}
                />
                <div
                    className="field__penalty-line"
                    style={{
                        borderTop: 'none',
                        top: 0,
                    }}
                />
                <div
                    className="field__penalty-line"
                    style={{
                        borderBottom: 'none',
                        bottom: 0,
                    }}
                />
                <div className="field__middle-line" />
                <div className="field__middle-circle" />
            </div>
            {showPositionAreas ? (
                <>
                    {positionList.map((position) => (
                        position.getArea()
                    ))}
                </>
            ) : null}
            <div className="field__drag-layer">
                {players.map((player) => (
                    <Player
                        key={player.id}
                        id={player.id}
                        name={player.name}
                        num={player.num}
                        x={player.position.x}
                        y={player.position.y}
                        mainColor={mainColor}
                        secondaryColor={secondaryColor}
                        numberColor={numberColor}
                        goal={player.goal || false}
                        yellowCard={player.yellowCard || false}
                        redCard={player.redCard || false}
                        setPlayerPosition={setPlayerPosition}
                    />
                ))}
            </div>
        </div>
    );
};

Field.propTypes = {
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    players: PropTypes.arrayOf(
        PropTypes.shape().isRequired,
    ).isRequired,
    mainColor: PropTypes.string.isRequired,
    secondaryColor: PropTypes.string.isRequired,
    numberColor: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    fieldType: PropTypes.string.isRequired,
    setPlayerPosition: PropTypes.func.isRequired,
    showPositionAreas: PropTypes.func.isRequired,
};

export default Field;
