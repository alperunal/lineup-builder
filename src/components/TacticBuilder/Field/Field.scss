.field {
    position: relative;
    height: 500px;
    width: 100%;
    background-color: #1b7e45;
    background-image: url('../../../assets/images/grass-tile.png');

    &__drag-layer {
        position: absolute;
        width: 100%;
        height: 100%;
        top: 0;
        left: 0;
    }

    &__grass-texture {
        position: absolute;
        background-repeat: repeat;
        background-size: 75px 75px;
        background-position: -20px -20px;
    }

    &__field-lines {
        width: 344px;
        height: 494px;
        border: solid 3px rgba(255, 255, 255, 0.5);
        position: relative;
    }

    &__goal-line {
        width: 16%;
        height: 6%;
        border: solid 3px rgba(255, 255, 255, 0.5);
        position: absolute;
        left: 50%;
        transform: translateX(-50%);
    }

    &__penalty-line {
        position: absolute;
        width: 44%;
        height: 14%;
        border: solid 3px rgba(255, 255, 255, 0.5);
        left: 0;
        right: 0;
        margin: auto;
    }

    &__middle-line {
        position: absolute;
        box-sizing: border-box;
        width: 100%;
        height: 0;
        border: solid 2px rgba(255, 255, 255, 0.5);
        top: 50%;
        transform: translateY(-50%);
        left: 0;
        right: 0;
    }

    &__middle-circle {
        position: absolute;
        box-sizing: border-box;
        width: 20%;
        height: 16%;
        border: solid 3px rgba(255, 255, 255, 0.5);
        top: 50%;
        transform: translateY(-50%);
        left: 0;
        right: 0;
        margin: auto;
        border-radius: 50%;
    }
}