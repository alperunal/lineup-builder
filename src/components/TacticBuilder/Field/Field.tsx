import React from 'react';
import Player from 'components/TacticBuilder/Player/Player';
import { IPlayer } from 'constants/model';
import { PositionArea } from 'constants/position';
import grassTile from 'assets/tiles/grass-tile.png';
import wetGrassTile from 'assets/tiles/wet-grass-tile.png';
import dryGrassTile from 'assets/tiles/dry-grass-tile.png';
import './Field.module.scss';

type Props = {
    width: number;
    height: number;
    players: IPlayer[];
    mainColor: string;
    secondaryColor: string;
    numberColor: string;
    fieldType: string;
    id: string;
    setPlayerPosition: (id: string, x: number, y: number) => void;
    showPositionAreas: boolean;
}

function getFieldStyle(fieldType: string) {
    switch (fieldType) {
    case 'hard':
        return 'none';
    case 'grass':
        return `url('${grassTile}')`;
    case 'dry-grass':
        return `url('${dryGrassTile}')`;
    case 'wet-grass':
        return `url('${wetGrassTile}')`;
    default:
        return `url('${grassTile}')`;
    }
}

const Field: React.FC<Props> = ({
    width, height, players, mainColor, secondaryColor, numberColor, id, fieldType, setPlayerPosition, showPositionAreas,
}: Props) => {
    const positionList = [
        new PositionArea('ST', '#4d4dff'),
        new PositionArea('WL', '#a862ea'),
        new PositionArea('WR', '#ff4dff'),
        new PositionArea('AMC', '#ff6680'),
        new PositionArea('ML', '#ff4d4d'),
        new PositionArea('MC', '#ea9a62'),
        new PositionArea('MR', '#ffff4d'),
        new PositionArea('DMC', '#c2c2ff'),
        new PositionArea('WBL', '#e1c9f8'),
        new PositionArea('FBL', '#ffc2ff'),
        new PositionArea('DC', '#ffb3bf'),
        new PositionArea('WBR', '#ffbebe'),
        new PositionArea('FBR', '#f6dbc6'),
        new PositionArea('GK', '#ffe7be'),
    ];

    return (
        <div
            className="field"
            style={{
                maxHeight: `${height}px`,
                maxWidth: `${width}px`,
                backgroundImage: getFieldStyle(fieldType),
            }}
            id={id}
        >
            <div className="field__field-lines">
                <div className="field__grass-texture" />
                <div
                    className="field__goal-line"
                    style={{
                        top: '-2px',
                    }}
                />
                <div
                    className="field__goal-line"
                    style={{
                        bottom: '-2px',
                    }}
                />
                <div
                    className="field__penalty-line"
                    style={{
                        borderTop: 'none',
                        top: 0,
                    }}
                />
                <div
                    className="field__penalty-line"
                    style={{
                        borderBottom: 'none',
                        bottom: 0,
                    }}
                />
                <div className="field__middle-line" />
                <div className="field__middle-circle" />
            </div>
            {showPositionAreas ? (
                <>
                    {positionList.map((position: PositionArea) => (
                        position.getArea()
                    ))}
                </>
            ) : null}
            <div className="field__drag-layer">
                {players.map((player) => (
                    <Player
                        key={player.id}
                        id={player.id}
                        name={player.name}
                        num={player.num}
                        x={player.position.x}
                        y={player.position.y}
                        mainColor={mainColor}
                        secondaryColor={secondaryColor}
                        numberColor={numberColor}
                        goal={player.goal || false}
                        yellowCard={player.yellowCard || false}
                        redCard={player.redCard || false}
                        setPlayerPosition={setPlayerPosition}
                    />
                ))}
            </div>
        </div>
    );
};

export default Field;
