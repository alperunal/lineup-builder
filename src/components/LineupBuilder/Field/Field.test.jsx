import React from 'react';
import { shallow } from 'enzyme';
import Field from './Field';

it('should render Field correctly', () => {
  const props = {
    width: 350,
    height: 500,
    players: [],
    mainColor: '#FFFFFF',
    secondaryColor: '#000000',
    numberColor: '#CCCCCC',
    fieldType: 'grass',
    id: '1',
  };
  const wrapper = shallow(<Field {...props} />);
  expect(wrapper).toMatchSnapshot();
});
