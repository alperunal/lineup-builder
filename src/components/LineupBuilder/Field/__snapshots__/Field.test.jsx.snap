// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Field correctly 1`] = `ShallowWrapper {}`;
